theano/gpuarray/opt.py

```diff
@@ -2069,7 +2085,7 @@ def provide_inputs(self, node, inputs):
                                          node.op.subsample,
                                          node.op.filter_dilation)
 
-            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[2:])
+            result[kshape] = theano.tensor.as_tensor_variable(node.op.kshp[-2:])
```

Comment:
`theano.gpuarray.tests.test_opt.Conv_opt_test`  tests the new optimisers I wrote, it does not test the default optimisers which are already there. The alternative optimisers for gradweight pass do not require the kshape argument, that is why these tests pass. The default gradweights optimisers ie `local_abstractconv_gw_cudnn` and `local_abstractconv3d_gradweights_gemm` will cause an error since they require the kshape argument